pkg/db/errors.go

```diff
@@ -18,3 +20,24 @@ func isDialError(err error) bool {
 	netError := &net.OpError{}
 	return errors.As(err, &netError) && netError.Op == "dial"
 }
+
+func handleSQLError(query string, err error, log logging.Logger, cmdType string) error {
+	if err == nil {
+		return nil
+	}
+
+	dbErrorsCounter.WithLabelValues(cmdType).Inc()
+	log.WithError(err).Error("SQL query failed with error")
+
+	// Each error that is added here should be updated also in controller.go:handleAPIError
+	if isUniqueViolation(err) {
+		err = ErrAlreadyExists
+	} else if pgxscan.NotFound(err) || errors.Is(err, pgx.ErrNoRows) {
+		log.Trace("SQL query returned no results")
+		err = ErrNotFound
+	} else {
+		err = fmt.Errorf("query %s: %w", query, err)
+	}
+
+	return err
```

Comment:
I'm a fan of early-exit, aka minimal indentation / `else`s.  It simplifies understanding of return flow.  In this case it would be
```suggestion
	if isUniqueViolation(err) {
		return ErrAlreadyExists
	}
	if pgxscan.NotFound(err) || errors.Is(err, pgx.ErrNoRows) {
		log.Trace("SQL query returned no results")
		return ErrNotFound
	}
	return fmt.Errorf("query %s: %w", query, err)
```

Also we should be careful about including queries in errors, in case they leak.